spec/support/comment_shared_examples.rb

```diff
@@ -74,6 +105,59 @@
         expect(page).not_to have_css "form#edit_comment_#{id}"
       end
     end
+
+    describe 'local caching', js: true do
+      let!(:original_page_with_comment_path) { page.current_path }
+
+      before do
+        find("div#comment_#{model.id}").hover
+        within "div#comment_#{model.id}" do
+          click_link 'Edit'
+          fill_in 'comment[content]', with: 'test comment edited'
+          sleep 1 # Needed for debounce function in local_auto_save.js
+          # click_button 'Update comment'
+        end
+      end
+
+      context 'when comment is not updated' do
+        it 'prefills textarea with cached value' do
+          visit root_path
+          visit original_page_with_comment_path
+          find("div#comment_#{model.id}").hover
+          within "div#comment_#{model.id}" do
+            click_link 'Edit'
+            content = page.find_field('comment[content]').value
+            expect(content).to eq 'test comment edited'
+          end
+        end
+
+        it 'does not save unsaved changes in database' do
+          expect(model.content).not_to eq 'test comment edited'
+        end
+      end
+
+      context 'when cancel is clicked' do
+        it 'clears cached value' do
+          visit root_path
+          visit original_page_with_comment_path
+          find("div#comment_#{model.id}").hover
+          within "div#comment_#{model.id}" do
+            click_link 'Edit'
+            click_link 'Cancel'
+          end
+
+          visit root_path
+          visit original_page_with_comment_path
```

Comment:
Is visiting root necessary or can we just re-navigate to `original_page_with_comment_path`?

Maybe even `page.reload`

The additional navigation to root_path feels weird and unnecessary. Same on [line #124](https://github.com/dradis/dradis-ce/pull/652/files#diff-ed90f32a3b3bcbd4317d9d0def453301R124)

Reply:
Ahh, i didn't even think about that, thanks! I used `page.driver.browser.navigate.refresh` instead of `page.reload` since `page.reload` doesn't work.